chia/wallet/util/peer_request_cache.py

```diff
@@ -13,18 +13,24 @@ class PeerRequestCache:
     _block_requests: LRUCache  # (start, end) -> RequestHeaderBlocks
     _ses_requests: LRUCache  # height -> Ses request
     _states_validated: LRUCache  # coin state hash -> last change height, or None for reorg
+    _timestamps: LRUCache  # block height -> timestamp
 
     def __init__(self):
         self._blocks = LRUCache(100)
         self._block_requests = LRUCache(100)
         self._ses_requests = LRUCache(100)
         self._states_validated = LRUCache(1000)
+        self._timestamps = LRUCache(1000)
 
     def get_block(self, height: uint32) -> Optional[HeaderBlock]:
         return self._blocks.get(height)
 
     def add_to_blocks(self, header_block: HeaderBlock) -> None:
         self._blocks.put(header_block.height, header_block)
+        if header_block.is_transaction_block:
+            assert header_block.foliage_transaction_block is not None
+            if self._timestamps.get(header_block.height) is None:
+                self._timestamps.put(header_block.height, header_block.foliage_transaction_block.timestamp)
```

Comment:
No change suggested for this PR, just a comment in context.

Makes me think of [`dict.setdefault()`](https://docs.python.org/3.10/library/stdtypes.html#dict.setdefault).  Not useful if the value takes relevant calculation, but otherwise `value = a_dict.setdefault(the_key, some_default)` is pretty clean.